client/src/components/Visualization/charts/ScatterPlot.js

```diff
@@ -0,0 +1,156 @@
+/**
+ * Description: exports a react/victory object of 
+ *              a scatterplot with the top X most prolific advertisers
+ *              across 2018, 2019, 2020, as computed in `aggregates`
+ *              firestore collection. 
+ * Date: 7/21/2020
+ * Author: Robbie Marcus
+ */
+
+import { database } from '../../../firebase/firebase';
+import React, { useState, useEffect } from 'react';
+import { VictoryTheme, 
+         VictoryScatter, 
+         VictoryChart, 
+         VictoryAxis, 
+         VictoryTooltip, 
+         VictoryLabel } from 'victory';
+
+/**
+ * Creates a hexidecimal color string from a string input. 
+ * Taken from here: `https://stackoverflow.com/a/16348977`.
+ * Loosely cleaned up for style and comments.
+ * This *should* return a valid hash, but if it doesn't,
+ * the browser will just display `#000`.
+ * @param {string} str input string 
+ * @return {string}
+ */
+function stringToColor(str) {
```

Comment:
Since this is just an arbitrary string, why aren't you just generating a random hex code? Scarlet has a method to do that in utils

Reply:
in addition, if your method is different from mine in Utils.js, just move this function there

Reply:
Did not realize that method existed! Neat, and have moved my method to be with it. 

Pros/cons of using `stringToColor`: 
pro: consistency without effort. Every time the user revisits/refreshes the chart with random color generator, all the bubbles will assume a different color. This could be confusing/misleading, i.e., did the data just change?, or, if it didn't, why did the chart change...? 

con: computationally more complex. The function is `O(n)` for string length `n`, random color is `O(1)`. Though practically none of the strings that I'd be calling it with are very large, don't think I've seen an advertiser with more than 30-40 characters in it. So it does incur a complexity cost but then again, not really.  

`m` strings, `O(mn)` cost to get all colors with my method. `O(m)` with scarlets method. But also I wouldn't want to use scarlets method without some kind of dict mapping advertisers to a color (would not want `adv_a 2019` to have a different color than `adv_a 2020`.) 

So additional client side space cost of `O(m)`, and complexity of adding that logic to the code. 

| Cost | Deterministic | Random | 
|------|--------------|----------|
|Comp| `O(mn)`        | `O(m)`    |
|(additional) space | `O(1)` | `O(m)`|